ocaml/parsing/parser.mly

```diff
@@ -1061,6 +1061,7 @@ The precedences must be listed from low to high.
           LBRACE LBRACELESS LBRACKET LBRACKETBAR LBRACKETCOLON LIDENT LPAREN
           NEW PREFIXOP STRING TRUE UIDENT
           LBRACKETPERCENT QUOTED_STRING_EXPR
+%nonassoc MATCH
```

Comment:
Can you add a CR-soon to the MATCH precedence as well? Given that we don't fully understand why giving this a high precedence fixes the issues, I'm concerned that we're breaking some other parsing. I'd like to revisit this later and see if we still need it.

Reply:
Added a CR-soon